apps/core/src/hooks/useGetRollingAverageApys.ts

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useMemo } from 'react';
+
+import { useGetValidatorsEvents } from './useGetValidatorsEvents';
+import { roundFloat } from '../utils/roundFloat';
+
+// recentEpochRewards is list of the last 30 epoch rewards for a specific validator
+// APY_e = (1 + epoch_rewards / stake)^365-1
+// APY_e_30rollingaverage = average(APY_e,APY_e-1,…,APY_e-29);
+
+const ROLLING_AVERAGE = 30;
+const DEFAULT_APY_DECIMALS = 4;
+
+// define the type parsedJson response
+type ParsedJson = {
+    commission_rate: string;
+    epoch: string;
+    pool_staking_reward: string;
+    pool_token_exchange_rate: {
+        pool_token_amount: string;
+        sui_amount: string;
+    };
+    reference_gas_survey_quote: string;
+    stake: string;
+    storage_fund_staking_reward: string;
+    tallying_rule_global_score: string;
+    tallying_rule_reporters: string[];
+    validator_address: string;
+};
+
+interface ApyGroups {
+    [validatorAddress: string]: number[];
+}
+
+export interface ApyByValidator {
+    [validatorAddress: string]: number;
+}
+
+export function useGetRollingAverageApys(numberOfValidators: number | null) {
```

Comment:
Feels like we could move this to indexer and have an api to fetch the APY per validator. (Not sure if possible at the moment though)

Reply:
Agreed, I will work with DivX for this